cabot/plugin_test_utils.py

```diff
@@ -0,0 +1,230 @@
+
+        transition_service(Service.PASSING_STATUS, Service.ERROR_STATUS)
+
+    You should probably at least test the following situations:
+    * passing -> warning
+    * passing -> error
+    * error -> passing
+    * error -> acked
+    """
+    def setUp(self):
+        update_alert_plugins()
+
+        self.user = User.objects.create(username='testuser', password='testuserpassword', email='test@test.test')
+        self.user.user_permissions.add(
+            Permission.objects.get(codename='add_service'),
+            Permission.objects.get(codename='add_httpstatuscheck'),
+            Permission.objects.get(codename='add_jenkinsstatuscheck'),
+            Permission.objects.get(codename='add_tcpstatuscheck'),
+        )
+        self.user.save()
+
+        self.jenkins_check = JenkinsStatusCheck.objects.create(
+            id=10101,
+            name='Jenkins Check',
+            created_by=self.user,
+            importance=Service.ERROR_STATUS,
+            max_queued_build_time=10,
+            max_build_failures=5
+        )
+
+        self.http_check = HttpStatusCheck.objects.create(
+            id=10102,
+            name='Http Check',
+            created_by=self.user,
+            importance=Service.CRITICAL_STATUS,
+            endpoint='https://google.com',
+            timeout=10,
+            status_code='200',
+            text_match=None,
+        )
+        self.tcp_check = TCPStatusCheck.objects.create(
+            id=10103,
+            name='TCP Check',
+            created_by=self.user,
+            importance=Service.ERROR_STATUS,
+            address='github.com',
+            port=80,
+            timeout=6,
+        )
+
+        self.service = Service.objects.create(id=2194, name='Service')
+
+        self.service.status_checks.add(
+            self.jenkins_check,
+            self.http_check,
+            self.tcp_check
+        )
+
+        self.service.users_to_notify.add(self.user)
+
+        self.duty_officer = User.objects.create(
+            username='dolores@affirm.com',
+            password='fakepassword',
+            email='dolores@affirm.com',
+            is_active=True
+        )
+        self.fallback_officer = User.objects.create(
+            username='teddy@affirm.com',
+            password='fakepassword',
+            email='teddy@affirm.com',
+            is_active=True
+        )
+
+        self.schedule = Schedule.objects.create(name='Test Schedule', fallback_officer=self.fallback_officer)
+        Shift.objects.create(user=self.duty_officer,
+                             start=timezone.now() - timezone.timedelta(minutes=1),
+                             end=timezone.now() + timezone.timedelta(hours=1),
+                             schedule=self.schedule)
+        self.service.schedules.add(self.schedule)
+
+        self.assertEquals(get_duty_officers(self.schedule), [self.duty_officer])
+        self.assertEquals(get_fallback_officers(self.schedule), [self.fallback_officer])
```

Comment:
Shouldn't have asserts in setUp()? would recommend writing an actual test (or just not doing this)